### PR TITLE
Collaborative editing

### PR DESCRIPTION
**Pull request by jimrandomh**

### CODE REVIEW

**Comment by Raemon** [comment]
Relatively minor note, but it's unintuitive to me that the "confirm" button isn't in the bottom-right of the sharing widget.

image

**Comment by Raemon** [comment]
Shouldn't be blocking, but, it'd be nice if Shared Posts showed up somewhere on my Profile page. (or something. They might be somewhere else and I haven't run into them yet)

**Comment by Raemon** [comment]
I notice that changes to "user can edit" don't take effect on my test-shared-user's page until I refresh. Shouldn't be blocking I think but does me I effectively can't stop users from editing once they've started, and probably a fairly major thing to fix soon.

**Review by Raemon — CHANGES_REQUESTED** [review]
It... looks like I can't type into comment fields! I've run `yarn install`, and `yarn rebuild-reinstall-ckeditor`

Commenting also isn't working on lessestwrong, where I have deployed the branch.

**Comment by jpaddison3** [comment]
Thanks for building this, Jim! UI comments:

 * I don't love the placement of the share button.
 * Sharing settings components is very in need of whitespace
 * So are the buttons for editing / deleting your own side-comment
 * Trying to submit a comment with ⌘-Enter submits the whole post form
 * The CollaborateOnPost route should set a white background
 * It's super cool that this is so realtime 😄
 * When you hover over another users cursor marker, it expands a line to the right that's clearly supposed to have their username, but the username is missing (it's in the html tho)
 * Another whitespace request: between the "by $USERNAME" and the circles telling you who's connected (this only appears when you're not the author) [and probably less between the title and "by $USERNAME"]
 * There should be a loading spinner when the editor is loading.

And some requests:

 * Could I get more context in the form of a PR description?
 * Can you say more about what's up with configuring ckeditor on the ckeditor side? I think our ckeditor keys are just copies of yours. (I don't remember, but have asked Oli.)
 * This appears to have the new user form page as well? That couldn't be factored out into a separate PR, could it?

**Comment by jpaddison3** [comment]
I think the UI should really warn you quite hard if you try to add an image to a non-online group. I know this is hard to do, but at the very least, hack it to say Online-Groups-Only as a title or something. (Maybe with a form group)

File: packages/lesswrong/components/posts/PostsPage/PostActions.tsx [hunk 1]
```diff
@@ -185,14 +200,7 @@ const PostActions = ({post, closeMenu, classes}: {
...
       <div className={classes.actions}>
-        { postCanEdit(currentUser,post) && <Link to={{pathname:'/editPost', search:`?${qs.stringify({postId: post._id, eventForm: post.isEvent})}`}}>
-          <MenuItem>
-            <ListItemIcon>
-              <EditIcon />
-            </ListItemIcon>
-            Edit
-          </MenuItem>
-        </Link>}
+        {editLink}
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/PostsPage/PostActions.tsx:203]: Good change.

**Pull request merged**